### PR TITLE
Remove the src/ folder to avoid using symlinks for custom code

### PR DESCRIPTION
**Pull request by titouanmathis**
This feature propose a revamp of the project's structure to avoid using symlinks for custom plugins, mu-plugins and themes as they are not reliable on all types of environment.

We take advantage of the ability of Wordpress to be installed in a separate repository to avoid overwriting the `wp-content/` folder on each `composer install` command.

See the following links for more details:
- https://github.com/johnpbloch/wordpress-core-installer
- https://composer.rarst.net/recipe/site-stack/

### CODE REVIEW

File: bin/source/wp-config-sample.php [hunk 1]
```diff
@@ -111,10 +111,8 @@
 define('WP_SITEURL', WP_HOME . getenv('APP_CORE'));
```
Comment (perruche) [inline bin/source/wp-config-sample.php]: `APP_CORE` has been removed from .env.example but is still use here

File: composer.json [hunk 2]
```diff
@@ -75,7 +70,7 @@
                 "type:wordpress-dropin"
             ]
         },
-        "wordpress-install-dir": "web"
+        "wordpress-install-dir": "web/app/"
```
Comment (perruche) [inline composer.json]: What are the thoughts behind naming the folder `app` 
Could this be name `wp` or `core` for more clarity ?

Comment (titouanmathis) [reply composer.json]: I took the `app` folder name from a similar setup I found as an example. But`wp` or `core` seem indeed better! I will change it for `wp`, I think it is the easiest to understand.

File: bin/source/wp-config-sample.php (hunk 1 above)
Comment (titouanmathis) [reply bin/source/wp-config-sample.php]: Nice catch! I think I edited the wrong `wp-config.php` 🤦‍♂

File: composer.json (hunk 2 above)
Comment (ptt-homme) [reply composer.json]: It seems reasonable to use `wp` as root for WordPress core. Just to know, will it be possible to take a bit of time on this PR or create a new one, to have a discuss on this project?

Comment (notjb) [reply composer.json]: i'd use `core`.
`wp` is too contextual to me

Comment (titouanmathis) [reply composer.json]: > It seems reasonable to use `wp` as root for WordPress core. Just to know, will it be possible to take a bit of time on this PR or create a new one, to have a discuss on this project?

Yes, with pleasure! I'd like to finish this as a kind of proof of concept on which we could discuss and iterate.

**Comment by titouanmathis** [comment]
This PR is obsolete.

**Pull request closed without merging**